### PR TITLE
feat: support request cancellation

### LINKED ISSUE #31: Support cancellation like Axios
With [google-api-nodejs-client](https://github.com/googleapis/google-api-nodejs-client)'s move from Axios to gaxios, the [cancellation feature](https://github.com/axios/axios#cancellation) has been lost. This turns out to be important for certain use cases. 

For example, the PubSub pull API has an option to return immediately or wait for some messages to arrive. Waiting for some messages is desirable if polling in a loop, so the loop doesn't poll overly frequently causing a high request volume and incurring costs for each request. But if the client decides to cancel the request, it may be stuck waiting for an indeterminate amount of time without the ability to cancel the request.

The ideal solution would be to support the same cancellation semantics as Axios, or some other cancellation API.

### PR DESCRIPTION
**Pull request by stephenplusplus**
Fixes #31 

This is the way to cancel a request through node-fetch: https://github.com/bitinn/node-fetch#request-cancellation-with-abortsignal

In theory, we could make it easier for the user by creating a controller for each request, and return a method like "request.cancel()". Or, not. Not is what this PR does.

### CODE REVIEW

**Comment by stephenplusplus** [comment]
Sure, that makes sense. Do you mean like this: https://github.com/axios/axios#cancellation?

**Comment by JustinBeckwith** [comment]
Exactly!

**Comment by stephenplusplus** [comment]
@JustinBeckwith So I know what we're going for with the whole cancel token approach... but can we just use this thing? I'm still stuck on how to recreate the dead proposal without taking a dependency or using the axios code... I think I either need an ELI5 or maybe pick up where you leave off if you want to get it started?
